openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/TypeGeneratorUtils.java

```diff
@@ -85,27 +85,28 @@
      * Get SchemaType object relevant to the schema given.
      *
      * @param schemaValue Schema object
+     * @param typeNameToken parameter name
      * @return Relevant SchemaType object
      */
-    public static TypeGenerator getTypeGenerator(Schema<?> schemaValue) {
+    public static TypeGenerator getTypeGenerator(Schema<?> schemaValue, IdentifierToken typeNameToken) {
```

Comment:
Not getting the reason to pass `typeNameToken` along. How does it affect to array length configuration ?

Reply:
yeah, it is right, the schema is enough for the drive to get the type of field.

But this is for generating the warning message more detailed full (as per discuss offline,  to provide warning message including details we are not going to support maxItem), by looking at schema we won't able to get enough details to populate the message with property or field it occurs in spec since it only includes the type of the given field or variable. 🙁 

Therefore I pass the type name here.  I have noticed that in a few other places, we are throwing common messages due to this matter, hope this would help to refactor those messages as well with fields. 
Since this is a warning and YAML can contain more fields with maxItem, hope we won't able to go with a common message like previous approaches.

Please make me correct, if I miss the way we can catch that detail through only looking at the schema? 🙂

Reply:
@lnash94 Then why don't we pass just the `typeName` instead of populating the ST token of the type name identifier? Passing the type name when creating a type generator instance makes more sense to me. @aneeshafedo @lnash94 WDYT?

Reply:
Or we can pass `Map.Entry<String, Schema> schema ` to the `TypeGenerator` without passing only the `schema.value`. Then `TypeGenerator` has access to both `schema.value` & `schema.key`. Just an idea.

Reply:
> @lnash94 Then why don't we pass just the `typeName` instead of populating the ST token of the type name identifier? Passing the type name when creating a type generator instance makes more sense to me. @aneeshafedo @lnash94 WDYT?

Here I passed the ST token by thinking in case we need this in future since we generate Syntax Tree. 🙂 

> Or we can pass `Map.Entry<String, Schema> schema ` to the `TypeGenerator` without passing only the `schema.value`. Then `TypeGenerator` has access to both `schema.value` & `schema.key`. Just an idea.
This is also nice since swagger provides a map with key-value pairs and we can pick one from them to pass one element of the map as key-value pair  🙂 

I'm afraid that if we hope to use this schema generator for other JSON and record/ type generation tasks, what would be the more sense and effective way to pass value to type Generator API, by considering these passing two parameters separately make sense to me 🤔 ? @NipunaRanasinghe @aneeshafedo  WDYT?

Reply:
+1 for both, but slightly prefer the 2 args constructor considering the readability and cleanliness

Reply:
+1